src/pint/models/model_builder.py

```diff
@@ -444,11 +471,17 @@ def choose_model(self, param_inpar):
         # the overall control parameters. This will get us the binary model name
         # build the base fo the timing model
         # pint_param_dict, unknown_param = self._pintify_parfile(param_inpar)
-        binary = param_inpar.get("BINARY", None)
-        if binary is not None:
+        if not force_binary_model:
+            binary = param_inpar.get("BINARY", None)
+        else:
+            binary = [force_binary_model]
```

Comment:
Maybe change this to an if statement?

Reply:
Looking at it, I need to rewrite this a bit. I don't know why I even keep track of `binary` here in this way. It's only used for the error message. Previously I needed it to keep track of the model, but I have been a bit sloppy here with the editing.

Reply:
Rewritten